### PR TITLE
Introduction to Buffers and Streams in Node.js

### PR DESCRIPTION
**Pull request by ReactifyStudio**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by ReactifyStudio** [comment]
This fixes #1254

**Comment by katungi** [comment]
@ReactifyStudio I have started the preliminary review

**Review by katungi — COMMENTED** [review]
Hello @ReactifyStudio , The grammar in the article is top notch and I like the flow of the article. Please @hectorkambow run a plagiarism check as I continue to the tech review

File: articles/node-buffer-streams/index.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+```javascript
+const Stream = require('stream');
+const readableStream = new Stream.Readable();
+
+readableStream.push('this is a readable stream');
+readableStream.push('Yet another data push to the readable stream');
+readableStream.push(null);
+```
+
+In the above code, we can see that readable streams are generators of data where we can write data using the `readableStream.push()` method. The `new Stream.Readable()` interface will initialize the `readableStream` object before pushing data into it using the `readableStream.push()` method. Passing a `null` in the `readableStream.push(null)` tells the consumer that our stream object is done outputting the data.
```
Comment (katungi) [inline articles/node-buffer-streams/index.md:150]: Please review this first statement

**Comment by hectorkambow** [comment]
[Introduction to Buffers and Streams in Node.js #1579.pdf](https://github.com/section-engineering-education/engineering-education/files/6103812/Introduction.to.Buffers.and.Streams.in.Node.js.1579.pdf)


@ReactifyStudio  
Upon running your article through our 3rd party plagiarism checker it raised a few red flags - and the % is a bit higher than we typically like to accept - please revisit and make sure we are using original and unique content. And adding sources wherever applicable. 👍

**Pull request closed without merging**